result.go

```diff
@@ -49,7 +48,9 @@ type Before struct {
 	Attachments   []Attachment   `json:"attachments,omitempty"`
 }
 
-type HasSteps interface {
+// This interface provides functions required to manipulate children step records, used in the result object and
+// step object for recursive handling
+type hasSteps interface {
```

Comment:
if you make it private, the comment is not necessary. It is necessary for public methods. Please have a look at https://blog.golang.org/godoc-documenting-go-code